core/src/main/java/com/amplifyframework/analytics/PropertyNames.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.analytics;
+
+import androidx.annotation.NonNull;
+
+import java.util.HashSet;
+import java.util.Iterator;
+
+/**
+ * PropertyNames holds the names of {@link Property} objects used in configuration calls.
+ *
+ * <pre>
+ *     PropertyNames propertyNames = PropertyNames.of("Height", "HairColor", "StarSign");
+ * </pre>
+ */
+public final class PropertyNames implements Iterable<String> {
+    private final HashSet<String> propertyNames;
```

Comment:
The convention in the rest of the project is to declare the field using its type of least specificity (Liskov Principle); here, this would be just `Set<String>`.